astropy/modeling/utils.py

```diff
@@ -493,3 +493,190 @@ def combine_labels(left, right):
+
+    t = np.arctan2(b, a * np.tan(theta))
+    dy = b * np.sin(t) * np.cos(theta) + a * np.cos(t) * np.sin(theta)
+    return np.abs([dx, dy])
+
+
+def render_model(model, image=None, coords=None, copy=True):
+    """
+    Evaluates a model on an input array. Evaluation is limited to
+    a bounding box if the `bounding_box` property is defined for
+    the model.
+
+    Parameters
+    ----------
+    model : Model
+            Model to be evaluated
+    image : array
+            Array on which the model is evaluated.
+    coords: list
+            List of coordinate arrays mapping to the image array,
+            such that `image[coords] == image`.
+    copy : bool, optional
+           Default is true. If False, image will be modified in place. 
+
+    Returns
+    -------
+    image : array
+            The evaluated model on the input image or coordinates. If 
+            no image or coordinates are input, the returned array is 
+            limited to the bounding_box of the model. The the bounding_box
+            is not defined, an image or coordinates must be passed.
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+            import numpy as np
+        import matplotlib.pyplot as plt
+
+        from astropy.modeling.models import Gaussian2D
+        from astropy.modeling.utils import render_model
+
+        imshape = (450, 600)
+        nsrc = 50
+        models = {}
+        models['x_mean'] = np.random.rand(nsrc) * imshape[1]
+        models['y_mean'] = np.random.rand(nsrc) * imshape[0]
+        models['amplitude'] = np.abs(np.random.randn(nsrc) + 10.)
+        models['x_stddev'] = np.abs(np.random.rand(nsrc) + 5)
+        models['y_stddev'] = models['x_stddev'] * (np.random.rand(nsrc) + 1)
+        models['theta'] = np.random.rand(nsrc) * np.pi
+
+        model_list = [Gaussian2D(x_mean=models['x_mean'][i], y_mean=models['y_mean'][i], 
+                                x_stddev=models['x_stddev'][i], y_stddev=models['y_stddev'][i],
+                                amplitude=models['amplitude'][i], theta=models['theta'][i])
+                                for i in range(nsrc)]
+
+
+        bb_image = np.zeros(imshape)
+        full_image = np.zeros(imshape)
+
+        for model in model_list:
+            render_model(model, bb_image, copy = False)
+            model.bounding_box = None
+            render_model(model, full_image, copy = False)
+
+        plt.figure(figsize=(12,8))
+        plt.imshow(bb_image, cmap='binary')
+        plt.colorbar()
+        plt.figure(figsize=(12,8))
+        plt.imshow(full_image - bb_image, cmap='binary', vmin=-1e-5, vmax=1e-5)
+        plt.colorbar(format='%.0e')
+        plt.show()
+    """
+
+    from .. nddata.utils import add_array, extract_array
+
+    bbox = model.bounding_box
+
+    if (coords is None) & (image is None) & (bbox is None):
+        raise AssertionError(
+            'If no bounding_box is set, coords or image must be input')
```

Comment:
This looks like a line break that was introduced by autopep8.
I'm not a big fan and would prefer to either violate the line length limit a bit:
```python
raise AssertionError('If no bounding_box is set, coords or image must be input')
```
or you use this format for things that don't fit on one line:
```python
raise AssertionError('If no bounding_box is set, '
                     'coords or image must be input')
```